### PR TITLE
[wip] upgrade Chef

### PR DESCRIPTION
**Pull request by srenatus**
Known TODOs:

* [ ] Upgrade Chef in oc_id
  <details>
    <summary>oc-id uses omniauth-chef, which also depends on `Chef ~> 12`; removing that requirements leads to a rack version incompatibility issue</summary>

  ```
  Bundler could not find compatible versions for gem "rack":
    In Gemfile:
      rails (= 4.2.8) was resolved to 4.2.8, which depends on
        actionpack (= 4.2.8) was resolved to 4.2.8, which depends on
          rack (~> 1.6)

      omniauth-chef was resolved to 0.3.0, which depends on
        chef (~> 13) was resolved to 13.0.113, which depends on
          chef-zero (>= 13.0) was resolved to 13.0.0, which depends on
            rack (~> 2.0)
  ```  
  :warning: this requires to upgrade to Rails 5.x.y; which bumping _maaany_ dependencies, so maybe it's better to
* [ ] Remove Chef dependency from oc-id -- it's only used in two places.

### CODE REVIEW

**Comment by srenatus** [comment]
needs a rebase after #1323.

File: omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+provides :chef_run
+actions :run
```
Comment (tas50) [inline omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb]: you don't need actions or default_action

File: omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+default_action :run
+
+property :run_list, :kind_of => String, :name_attribute => true
+property :cookbook_path
+property :included_attrs, :kind_of => Array, :default => []
```
Comment (tas50) [inline omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb]: You don't need the kind_of part. Just the ruby type

File: omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
+provides :chef_run
+actions :run
+default_action :run
+
+property :run_list, :kind_of => String, :name_attribute => true
```
Comment (tas50) [inline omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb]: This should be name_property not name_attribute

**Comment by markan** [comment]
Closed by #1506

**Pull request closed without merging**